cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1344 @@
+				err = UpdateCheckItem(a.CheckItems[z].ID, a.CheckItems[z].Name, a.CheckItems[z].State)
+			}
+		}
+	}
+	log.Println(errMap)
+	log.Printf("Following are the exchanges that require updates: %v\n", resp)
+	unsup, err := CheckMissingExchanges(fileName)
+	log.Printf("Following are the exchanges that are supported by GCT but not by apichecker: %v", unsup)
+	return ioutil.WriteFile(fileName, file, 0770)
+}
+
+// CheckChangeLog checks the exchanges which support changelog updates.json
+func CheckChangeLog(htmlData HTMLScrapingData) (string, error) {
+	var dataStrings []string
+	var dataString string
+	var err error
+	switch htmlData.Path {
+	case pathBTSE:
+		dataStrings, err = HTMLScrapeBTSE(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitfinex:
+		dataStrings, err = HTMLScrapeBitfinex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathBitmex:
+		dataStrings, err = HTMLScrapeBitmex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathANX:
+		dataStrings, err = HTMLScrapeANX(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathPoloniex:
+		dataStrings, err = HTMLScrapePoloniex(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathIbBit:
+		dataStrings, err = HTMLScrapeItBit(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBTCMarkets:
+		dataStrings, err = HTMLScrapeBTCMarkets(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathEXMO:
+		dataStrings, err = HTMLScrapeExmo(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitstamp:
+		dataStrings, err = HTMLScrapeBitstamp(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathHitBTC:
+		dataStrings, err = HTMLScrapeHitBTC(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathBitflyer:
+		dataStrings, err = HTMLScrapeBitflyer(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathLakeBTC:
+		dataStrings, err = HTMLScrapeLakeBTC(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathKraken:
+		dataStrings, err = HTMLScrapeKraken(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathAlphaPoint:
+		dataStrings, err = HTMLScrapeAlphaPoint(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataStrings[0], nil
+	case pathYobit:
+		dataStrings, err = HTMLScrapeYobit(htmlData)
+		if err != nil {
+			return "", err
+		}
+	case pathLocalBitcoins:
+		dataString, err = HTMLScrapeLocalBitcoins(htmlData)
+		if err != nil {
+			return "", err
+		}
+		return dataString, nil
+	default:
+		dataStrings, err = HTMLScrapeDefault(htmlData)
+		if err != nil {
+			return "", err
+		}
+	}
+	switch htmlData.Path {
+	case pathOkCoin, pathOkex:
```

Comment:
You have a htmlData.Path just above this. Why not just use that? If its because you're wanting to use default, you could just do:
```
case pathOkCoin, pathOkex:
dataStrings, err = HTMLScrapeDefault(htmlData)
		if err != nil {
			return "", err
		}
for x := range dataStrings {
			if len(dataStrings[x]) != 10 {
				tempStorage := strings.Split(dataStrings[x], "-")
				dataStrings[x] = fmt.Sprintf("%s-0%s-%s", tempStorage[0], tempStorage[1], tempStorage[2])
			}
		}
```